### PR TITLE
Added provenance hash functionality

### PR DESCRIPTION
**Pull request by tomvanleeuwennl**
A provenance hash is a way to make sure the collection is not tampered with during or after the minting period. Before minting starts you can provide the hashing list at your NFT website. The list consists of edition numbers and the corresponding derived unique hash. This list itself is also hashed by concatenating all the image hashes in the order of the edition numbers. So this list ensures there is no possibility to change images or the order of the collection.

More info about why to use provenance hash here:
https://medium.com/coinmonks/the-elegance-of-the-nft-provenance-hash-solution-823b39f99473

This provenance hash can then be saved into a NFT contract before the minting process starts. This ensures that the collection cannot be changed without changing the provenance hash.

### CODE REVIEW

**Comment by adamnooh** [comment]
Can u contact me at 864 7200913  please 🙏 important tax me first if I
didn't answer thanks I have couple questions please

**Comment by tomvanleeuwennl** [comment]
What do you want to know?

**Comment by jasan-s** [comment]
@tomvanleeuwennl , please add a quick summary in PR notes above the motivation for the PR i.e why provenance hash is needed and used.

**Comment by tomvanleeuwennl** [comment]
Ok! I am going to do that. I can also change it to a utility function. What is preferred?

**Comment by vizhang** [comment]
hey @tomvanleeuwennl this is an awesome consideration. I read the article you linked and it also mentions how provenance hash solves for "NFT projects should not release the final Token URI until the final NFT has been minted, to prevent savvy snipers from acquiring the best ones in the collection". Based on how HashedLipsEngine is implemented though, does the provenance hash solve for this?

File: utils/updateProvenanceHash.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+const fs = require("fs");
+const path = require("path");
+const crypto = require('crypto');
+const isLocal = typeof process.pkg === "undefined";
+const basePath = isLocal ? process.cwd() : path.dirname(process.execPath);
+
+const { baseUri } = require("../src/config.js");
+var provenanceHashList = [];
+
+const generatProvenance = (_provenanceHashList) => {
```
Comment (kwokwilliam) [inline utils/updateProvenanceHash.js:12]: (Apologies if I'm overstepping any bounds, maintainers! Don't see any docs about code reviewing, and I just took a look through some of the open PRs)

Small typo

```suggestion
const generateProvenance = (_provenanceHashList) => {
```

Also this and the `generateProvenance` below this are pretty similar, I think it'd be worth combining the functionality.

**Comment by richTheCreator** [comment]
This was exactly what I was looking for! Curious why this was closed with out being merged?

**Comment by shawwwwwwww** [comment]
Hey, this PR got merged, but seemed extremely valuable. Actually I was going to PR in the same feature :D
Can we reopen and merge?

**Pull request closed without merging**